### PR TITLE
feat(rust, python): Pushdown `is_in` to pyarrow dataset

### PR DESCRIPTION
**Pull request by chitralverma**
closes https://github.com/pola-rs/polars/issues/8887

### CODE REVIEW

**Comment by chitralverma** [comment]
@stinodego @ritchie46  please have a look

File: polars/polars-lazy/polars-plan/src/logical_plan/pyarrow.rs [hunk 1]
```diff
@@ -16,6 +16,11 @@ pub(super) fn predicate_to_pa(predicate: Node, expr_arena: &Arena<AExpr>) -> Opt
         }
         AExpr::Column(name) => Some(format!("pa.dataset.field('{}')", name.as_ref())),
         AExpr::Alias(input, _) => predicate_to_pa(*input, expr_arena),
+        AExpr::Literal(LiteralValue::Series(s)) => s
```
[the remark this answers is not in the document]
Comment (ritchie46) [reply polars/polars-lazy/polars-plan/src/logical_plan/pyarrow.rs:19]: No, it should return None. In that case we will not push the predicate to pyarrow but do it after the scan.

File: polars/polars-lazy/polars-plan/src/logical_plan/pyarrow.rs [hunk 2]
```diff
@@ -24,9 +44,9 @@ pub(super) fn predicate_to_pa(predicate: Node, expr_arena: &Arena<AExpr>) -> Opt
                 AnyValue::Boolean(val) => {
                     // python bools are capitalized
                     if val {
-                        Some("True".to_string())
+                        Some("pa.compute.scalar(True)".to_string())
```
Comment (chitralverma) [inline polars/polars-lazy/polars-plan/src/logical_plan/pyarrow.rs:47]: This change was required as returning `True`/ `False` was breaking currently as it gets evaluated to a bool. pyarrow requires an expression, not a bool

File: polars/polars-lazy/polars-plan/src/logical_plan/pyarrow.rs [hunk 3]
```diff
@@ -14,8 +16,26 @@ pub(super) fn predicate_to_pa(predicate: Node, expr_arena: &Arena<AExpr>) -> Opt
                 None
             }
         }
-        AExpr::Column(name) => Some(format!("pa.dataset.field('{}')", name.as_ref())),
+        AExpr::Column(name) => Some(format!("pa.compute.field('{}')", name.as_ref())),
```
Comment (chitralverma) [inline polars/polars-lazy/polars-plan/src/logical_plan/pyarrow.rs:19]: sync up with the latest pyarrow docs. pc is used in places of pa.ds

File: polars/polars-lazy/polars-plan/src/logical_plan/pyarrow.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (chitralverma) [reply polars/polars-lazy/polars-plan/src/logical_plan/pyarrow.rs]: took the above changes

**Review by ritchie46 — COMMENTED** [review]
The changes look great. Can you add a test for this on the python side? Then its good to go!

**Comment by alippai** [comment]
Not sure if it's relevant, but I find the current dataset filter a bit confusing: https://github.com/apache/arrow/issues/34727

**Pull request merged**